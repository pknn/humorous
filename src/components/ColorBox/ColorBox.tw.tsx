import tw from 'tailwind-styled-components'

export const Box = tw.div`
  w-full
  h-1/5

  first:rounded-t-lg
  last:rounded-b-lg

  md:w-1/5
  md:h-full

  md:first:rounded-l-lg
  md:first:rounded-tr-none

  md:last:rounded-r-lg
  md:last:rounded-bl-none
`
