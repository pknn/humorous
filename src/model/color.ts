export type ColorValue = number
export type ColorType = 'RGB' | 'HSL'

export interface Color {
  type: ColorType
  components: ColorValue[]
}
