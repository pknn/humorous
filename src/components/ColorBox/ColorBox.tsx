import { useMemo } from 'react'
import { Color } from '../../model/color'

export const getColorProperty = ({ type, components }: Color) =>
  `${type.toLowerCase()}(${components.join(',')})`

interface ColorBoxProps {
  color: Color
}

export const ColorBox = ({ color }: ColorBoxProps) => {
  const colorProperty = useMemo(() => getColorProperty(color), [color])

  return (
    <div style={{ background: colorProperty }} className="w-full h-full"></div>
  )
}
