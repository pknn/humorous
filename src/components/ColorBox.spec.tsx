import { describe, it, expect } from 'vitest'
import { shallow } from 'enzyme'
import { Color } from '../model/color'
import { ColorBox, getColorProperty } from './ColorBox'

const setup = (color: Color) => shallow(<ColorBox color={color} />)

type TestCase = Color[]
const testCases: TestCase = [
  { type: 'RGB', components: ['255', '255', '255'] },
  { type: 'HSL', components: ['240', '10%', '80%'] },
]

describe('<ColorBox />', () => {
  it('should render correctly', () => {
    const component = setup({ type: 'RGB', components: ['255', '255', '255'] })

    expect(component).toMatchSnapshot()
  })

  it.each(testCases)(
    'should render with correct color value when color is %s',
    (color) => {
      const component = setup(color)

      expect(component.prop('style')).toContain({
        background: getColorProperty(color),
      })
    },
  )
})
