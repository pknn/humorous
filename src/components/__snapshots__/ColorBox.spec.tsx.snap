// Vitest Snapshot v1

exports[`<ColorBox /> > should render correctly 1`] = `ShallowWrapper {}`;
